### DISCUSSION TITLE
Leak memory aiorts

### DISCUSSION THREAD

**Discussion by Plachey** [aiortc/aiortc#736] [Q&A]
Hi everyone. I have a trouble after start slixmpp.ClientXMPP video file using aiortc here 
I think problem in this line but can't to fix it:   frame = next(container.decode(*streams))
 Top 5 largest allocating locations (by size):
        - player_worker:/home/pl/PycharmProjects/creater_user_testing/venv/lib/python3.8/site-packages/aiortc/contrib/media.py:111 -> 6.881GB
        - encode:/home/pl/PycharmProjects/creater_user_testing/venv/lib/python3.8/site-packages/aiortc/codecs/vpx.py:280 -> 560.069MB
        - encode:/home/pl/PycharmProjects/creater_user_testing/venv/lib/python3.8/site-packages/aiortc/codecs/vpx.py:282 -> 126.667MB
        - player_worker:/home/pl/PycharmProjects/creater_user_testing/venv/lib/python3.8/site-packages/aiortc/contrib/media.py:132 -> 67.599MB
        - __init__:/home/pl/PycharmProjects/creater_user_testing/venv/lib/python3.8/site-packages/aiortc/rate.py:451 -> 5.250MB

Who else have the same problem? Please can advice how to fix it

**Comment by jlaine** [comment]
I have no idea what slimxpp is, but regarding aiortc: make sure you're consuming *all* the frames you receive like in the examples. Otherwise, yes memory consumption will grow.

**Comment by david-goldb** [comment]
Hey @Plachey,
Did you fix this?
I consume all tracks, but still RAM is accumulating due to the allocation of:
home/pl/PycharmProjects/creater_user_testing/venv/lib/python3.8/site-packages/aiortc/codecs/vpx.py

**Comment by flann-mc** [comment]
This rate.py file also grows in memory for me, but only if the peerconnection is closed before the stream is established. It's causing a huge slowdown of the application over time. The tracks in the peerconnection are not being properly deleted when `peerconnection.close()` is called.

**Reply by PerryDP** [reply]
> This rate.py file also grows in memory for me, but only if the peerconnection is closed before the stream is established. It's causing a huge slowdown of the application over time. The tracks in the peerconnection are not being properly deleted when `peerconnection.close()` is called.

My requirement is to switch streams without shutting down the PC
```

async def add_tracks(video):
    global pc, current_video_player
    old = current_video_player
    current_video_player = MediaPlayer(video,loop=True)
    for sender in pc.getSenders():
        if sender.track.kind == "video":
            sender.replaceTrack(current_video_player.video)
        if sender.track.kind == "audio":
            sender.replaceTrack(current_video_player.audio)
    if old:
        old.video.stop()
        old.audio.stop()
        del old
        gc.collect()

```


When I only have one mediaplayer object from start to finish, memory usage is normal. After switching streams multiple times, memory usage surges and continues to grow until the program is killed by the system. The occurrence time and number of switches for memory surges are very random, but usually after three switches. Even if I manually close old media streams, delete references, and perform garbage collection, it will not work
I am eager to know how to solve it
